next-docs/public/examples/snackbarTW/Queue.tsx

```diff
@@ -0,0 +1,90 @@
+import { Snackbar, Button } from '@heathmont/moon-core-tw';
+import { useState, useCallback } from 'react';
+
+type NotificationType = {
+  message: string;
+  type: string;
+};
+
+function useQueueState<NotificationType>(initialList: NotificationType[]): [
+  NotificationType[],
+  {
+    dequeue: () => NotificationType | undefined;
+    enqueue: (item: NotificationType) => number;
+    length: number;
+    peek: () => NotificationType | undefined;
+  }
+] {
+  const [list, setList] = useState<NotificationType[]>([...initialList]);
+  const enqueue = useCallback(
+    (item: NotificationType) => {
+      const newList = [...list, item];
+      setList(newList);
+      return newList.length;
+    },
+    [list]
+  );
+
+  const dequeue = useCallback(() => {
+    if (list.length > 0) {
+      const firstItem = list[0];
+      setList([...list.slice(1)]);
+      return firstItem;
+    } 
+    return undefined
+  }, [list]);
+
+  const peek = useCallback(() => {
+    if (list.length > 0) {
+      return list[0];
+    }
+    return undefined;
+  }, [list]);
+
+  const notifications = {
+    dequeue,
+    enqueue,
+    length: list.length,
+    peek,
+  };
+
+  return [list, notifications];
+}
+
+ 
+const Example = () => {
+  const [notificationList, notifications] = useQueueState([] as NotificationType[]);
+  const [isOpen, setIsOpen] = useState(false)
```

Comment:
i think setOpen and isOpen is replaced with list.length > 0 (or length > 0 or peek !== null).
also [list, notifications] should be probably named [list, options], because it has a set of functions to use